src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -875,6 +874,47 @@ private List<DataProviderMetadata> GetDataProvidersMetadata(DataProviders.DataPr
                         }
                     }
                 }
+                else if (dataProvider.FieldType.UnderlyingSystemType.Name.StartsWith("Func`2", StringComparison.OrdinalIgnoreCase))
```

Comment:
Partners will be using MdmGeneric. Can you find a way to also have this benefit them too?

Reply:
We need to come up with a better way of doing this for partners. Doing it the reflection way is trickier and dirtier as the MdmGeneric expects GenericMdmDataProviderConfiguration parameter and I cannot find an easy way via reflection to get that parameter. We need to think of that provider differently. Maybe as a separate PR altogether.

Reply:
Okay, lets just make sure we create a WI for this so this doesn't slip through the cracks

Reply:
Opened 1215 to track this.